File: server/src/main/java/com/exam/examMS/service/QuestionService.java
package com.exam.examMS.service;

import java.util.List;
import java.util.Set;

import com.exam.examMS.entity.exam.Question;
import com.exam.examMS.entity.exam.Quiz;

public interface QuestionService {

	public Question addQuestion(Question question);
	public Question updateQuestion(Question question);
	public Set<Question> getQuestion();
	public Question getQuestion(Long questionId);
	public Set<Question> getQuestionsOfQuiz(Quiz quiz);
	public void deleteQuestion(Long questionId);
	public List<Question> getQuestionsForEval(Set<Long> questionIdList);
}

File: client/src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private _http:HttpClient) { }

  public getQuestions(id:any)
  {
    return this._http.get(`${baseUrl}/question/quiz/all/${id}`); 
  }

  public getQuestionsForQuiz(id:any)
  {
    return this._http.get(`${baseUrl}/question/quiz/${id}`); 
  }

  public addQuestion(question:any)
  {
    return this._http.post(`${baseUrl}/question/`,question); 
  }

  deleteQuestion(id:any)
  {
    return this._http.delete(`${baseUrl}/question/${id}`); 
  }

  updateQuestion(question:any)
  {
    return this._http.put(`${baseUrl}/question/`,question); 
  }

  public getQuestion(id:any)
  {
    return this._http.get(`${baseUrl}/question/${id}`); 
  }

  public evalQuiz(questions:any)
  {
    return this._http.post(`${baseUrl}/question/eval-quiz`,questions);
  }
}
